docs/testing.md

```diff
@@ -0,0 +1,61 @@
+# Testing Storefront
```

Comment:
Shameless plug but I think this list needs some [automated todos.](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/docs/testing/cart-checkout/coupons.md)

Reply:
Feel free to submit a PR! Though I would caution against relying on these items – they are intended to spark good holistic testing of the product, rather than prescribe all the required things to tick off to guarantee quality :)